pkg/components/contour/component.go

```diff
@@ -80,6 +83,16 @@ func (c *component) RenderManifests() (map[string]string, error) {
 		return nil, errors.Wrap(err, "failed to walk assets")
 	}
 
+	// Set default service type to 'LoadBalancer'.
+	if c.ServiceType == "" {
```

Comment:
Why not in `newComponent`, so it gets overwritten by HCL if the field is set?

Reply:
Didn't realize we could do that. Fixed.
NOTE: This kind of patterns we can (and probably should) enforce for all components :shrug: